### PR TITLE
MGMT-15271: get cmdline & set vendor

### PR DESCRIPTION
**Pull request by AmadeusPodvratnik**
With the support of zVM in Assisted Installer, the necessary kernel arguments to boot a zVM nodes need to be considered. During the first boot of a zVM node a parm file will be provided to activate the necessary devices or specify a static IP w/o nmstate. This parm file looks like:

rd.neednet=1 console=ttysclp0 coreos.live.rootfs_url=http://172.23.236.156:8080/assisted-installer/rootfs.img ip=10.14.6.3::10.14.6.1:255.255.255.0:master-0.boea3e06.lnxero1.boe:encbdd0:none nameserver=10.14.6.1 ip=[fd00::3]::[fd00::1]:64::encbdd0:none nameserver=[fd00::1] zfcp.allow_lun_scan=0 rd.znet=qeth,0.0.bdd0,0.0.bdd1,0.0.bdd2,layer2=1 rd.dasd=0.0.5235 rd.dasd=0.0.5236 random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal console=tty1 console=ttyS1,115200n8

Nevertheless the cmdline will be taken independently of the architecture. For s390x gwh is returning for getProduct():
product family=unknown name=unknown vendor=unknown sku=unknown version=unknown
and need to be set to the correct values. To do that you need to parse:
cat /proc/cpuinfo |grep vendor
vendor_id    : IBM/S390
sudo cat /proc/sysinfo |grep 'VM.*Control Program'
VM00 Control Program: z/VM  7.2.0 
For KVM following values are returned:
cat /proc/sysinfo |grep 'VM.*Control Program'
VM00 Control Program: KVM/Linux

List all the issues related to this PR

    [x ] New Feature
          Enhancement
          Bug fix
          Tests
          Documentation
          CI/CD 

How was this code tested?

       assisted-test-infra environment
       dev-scripts environment
[x ] Reviewer's test appreciated
       Waiting for CI to do a full test run
[x ]  Manual (Elaborate on how it was tested) 
        - Using boot_test.go
      No tests needed

### CODE REVIEW

File: go.sum
Comment (omertuc) [inline go.sum:1]: Was this committed on purpose?

Comment (AmadeusPodvratnik) [reply go.sum:1]: I generated the project and go.sum was adapted, too. So I though that adding go.sum (if I run skipper) is needed. If not I will remove the file.

Comment (omertuc) [reply go.sum:1]: I don't expect changes to `go.sum` unless we also have changes to `go.mod`

File: src/inventory/boot.go [hunk 1]
```diff
@@ -41,10 +41,19 @@ func (b *boot) getCurrentBootMode() (ret string) {
...
+		return string(cmdline)
+	}
+	return ""
+}
+
 func (b *boot) getBoot() *models.Boot {
 	ret := models.Boot{
 		CurrentBootMode: b.getCurrentBootMode(),
 		PxeInterface:    b.getPxeInterface(),
+		CommandLine:     b.getCommandLine(),
```
Comment (omertuc) [inline src/inventory/boot.go:56]: This field doesn't exist in your PR. I think you need to bump the service models to the latest version to include your service changes.

File: src/inventory/boot.go [hunk 2]
```diff
@@ -41,10 +41,19 @@ func (b *boot) getCurrentBootMode() (ret string) {
 	return
 }
 
+func (b *boot) getCommandLine() string {
+    cmdline, err := b.dependencies.ReadFile("/proc/cmdline")
+    if err == nil {
```
Comment (omertuc) [inline src/inventory/boot.go:46]: The brain of go devs is hardwired to skim `if err =...` as `if err != nil` and not `if err == nil`, please swap the condition

**Comment by omertuc** [comment]
The files are not formatted properly, please use `gofmt`

**Comment by AmadeusPodvratnik** [comment]
@omertuc Due to the merge conflicts should I remove all other changes (beside assisted-services) which where introduced during the make and generate?

**Comment by omertuc** [comment]
> @omertuc Due to the merge conflicts should I remove all other changes (beside assisted-services) which where introduced during the make and generate?

Not sure I follow. Please rebase off of latest master branch to resolve conflicts

**Comment by omertuc** [comment]
Please do a rebase rather than a merge so you only have one commit

**Comment by AmadeusPodvratnik** [comment]
> Please do a rebase rather than a merge so you only have one commit

Sorry for the inconvenience but still struggling with go.mod, go.sum and the modules.txt. I run go mod tidy and go mod vendor to fix the issues on my side (the result is the new go.mod and go.sum files). Nevertheless I'm seeing in the tests the failures:
 	github.com/openshift/assisted-service/models: is marked as replaced by github.com/openshift/assisted-service/models@v0.0.0-20231010223859-e350e5ce7b20 in vendor/modules.txt, but replaced by github.com/openshift/assisted-service/models@v0.0.0-20231102123309-91b2f5ef1d69 in go.mod
	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
make: *** [Makefile:44: build-agent] Error 1 

I saw the same errors on my side but a go mod tidy and go mod vendor fixed them. So not sure how to solve this.

**Comment by AmadeusPodvratnik** [comment]
Redo changes in correct order. New PR will be opened.

**Pull request closed without merging**